alertmanager/alert.go

```diff
@@ -0,0 +1,31 @@
+package alertmanager
+
+// Alert is an Alertmanager alert.
+type Alert struct {
+	// Annotations are arbitrary key-value pairs.
+	Annotations map[string]string `json:"annotations"`
+
+	// Labels are key-value pairs that can be used to group and filter alerts.
+	Labels map[string]string `json:"labels"`
+}
+
+// NewAlert creates a new Alert with initialized maps.
+func NewAlert() *Alert {
```

Comment:
The WithLabel and WithAnnotation funcs imply the functional options pattern, but you haven't wired the Alert constructor up for that. Please make them functional options!